AnnoDesigner/AnnoCanvas.xaml.cs

```diff
@@ -1750,6 +1759,11 @@ protected override void OnMouseDown(MouseButtonEventArgs e)
                 //DragAll, so we fire it here instead, to prevent objects being incorrectly represented within the QuadTree.
                 if (CurrentMode == MouseMode.DragSelection)
                 {
+                    RegisterUndoableOperation(new MoveOperation()
```

Comment:
Maybe it is possible to move most of the logic to an injectible class like `UndoManager` or `UndoHelper` which implements a corresponding interface.
The target has to be to have as little as possible logic as part of the canvas.